plugins/modules/zos_script.py

```diff
@@ -0,0 +1,416 @@
+      - If C(encoding) is not provided, the module determines which local
+        and remote charsets to convert the data from and to.
+    type: dict
+    required: false
+    suboptions:
+      from:
+        description:
+          - The encoding to be converted from.
+        required: true
+        type: str
+      to:
+        description:
+          - The encoding to be converted to.
+        required: true
+        type: str
+  executable:
+    description:
+      - Path of an executable in the remote machine to invoke the
+        script with.
+      - When not specified, the system will assume the script is
+        interpreted REXX and try to run it as such. Make sure to
+        include a comment identifying the script as REXX at the
+        start of the file in this case.
+    type: str
+    required: false
+  remote_src:
+    description:
+      - If set to C(false), the module will search the script in the
+        controller.
+      - If set to C(true), the module will search the script in the
+        remote machine.
+    type: bool
+    required: false
+  removes:
+    description:
+      - Path to a file in the remote machine. If it does not exist, the
+        script will not be executed.
+    type: str
+    required: false
+  tmp_path:
+    description:
+      - Path in the remote machine where local scripts will be
+        temporarily copied to.
+      - When not specified, the module will copy local scripts to
+        the default temporary path for the user.
+      - If C(tmp_path) does not exist in the remote machine, the
+        module will not create it.
+    type: str
+    required: false
+
+extends_documentation_fragment:
+  - ibm.ibm_zos_core.template
+
+notes:
+  - When executing local scripts, temporary storage will be used
+    on the remote z/OS system. The size of the temporary storage will
+    correspond to the size of the file being copied.
+  - Execution permissions for the group assigned to the script will be
+    added to remote scripts. The original permissions for the script will
+    be restored by the module before the task ends.
+  - If executing REXX scripts, make sure to include a newline character on
+    each line of the file. Otherwise, the interpreter may fail and return
+    error C(BPXW0003I).
+  - For supported character sets used to encode data, refer to the
+    L(documentation,https://ibm.github.io/z_ansible_collections_doc/ibm_zos_core/docs/source/resources/character_set.html).
+  - This module uses L(zos_copy,./zos_copy.html) to copy local scripts to
+    the remote machine.
+  - L(zos_copy,./zos_copy.html) uses SFTP (Secure File Transfer Protocol)
+    for the underlying transfer protocol; Co:Z SFTP is not supported. In
+    the case of Co:z SFTP, you can exempt the Ansible userid on z/OS from
+    using Co:Z thus falling back to using standard SFTP.
+  - This module executes scripts inside z/OS UNIX System Services. For
```

Comment:
This is a good idea to point users to the tso module, it might be a good idea to open an issue there to added examples on how to do this. I originally thought we would cover data sets containing rexx but we can talk about where that best fits, maybe a in the tso module, new rexx module or here.